docs/source/reference/cli.rst

```diff
@@ -79,13 +79,43 @@ by passing ``--skip-config`` flag to the CLI as shown below:
 Authentication and auth token caching
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-If you specify username and password as authentication credentials in the
-configuration file, the CLI will try to use those credentials to authenticate and
-retrieve an auth token.
+If you don't wish to store your password in plain-text as shown in the previous section,
+the ``st2 login`` command offers an alternative. Similar to ``st2 auth``, you must provide your
+username and password:
```

Comment:
It's maybe also worth adding a note that since this command doesn't store password by default in the config and it relies on a cached auth token it will only work until the token expires (which is 24 hours by default, IIRC).

After that, the user will need to re-login. And / or perhaps also document the `--write-password` option.